### PR TITLE
Error recorder

### PR DESCRIPTION
**Pull request by ac-rappi**
Capabilities to differentiate errors for the cb counts

### CODE REVIEW

**Review by fabioboterolyons — APPROVED** [review]

**Review by fabioboterolyons — APPROVED** [review]

**Review by kolmant — APPROVED** [review]

**Comment by kolmant** [comment]
No sé si por lineamientos de ellos se deba tocar la documentación también con el cambio

**Review by amoto — APPROVED** [review]

**Review by moisesmorillo — APPROVED** [review]

File: README.md [hunk 1]
```diff
@@ -56,6 +57,11 @@ type Settings struct {
   If `ReadyToTrip` is `nil`, default `ReadyToTrip` is used.
   Default `ReadyToTrip` returns true when the number of consecutive failures is more than 5.
 
+- `RecordError` is called when a non-nil error is returned by the function, and intends to differentiate errors that should be used for tripping the circuit or not.
+  If `RecordError` returns true, `CircuitBreaker` will not consider the operation as successful
+	If `RecordError` is `nil`, default `RecordError` is used.
+  Default `RecordError` returns true
```
Comment (zeitos) [inline README.md]: Can you clarify that if returns false, then the operation will be unsuccessful ?

**Comment by SebastianOsuna** [comment]
👍

File: README.md (hunk 1 above)
Comment (ac-rappi) [reply README.md]: Default `RecordError` considers all errors should be recorded

☝️ Do you think that is clear?

Comment (zeitos) [reply README.md]: clarifies the logic behind `defaultRecordError `  but this text defines the RecordError semantic, so I'd suggest being explicit. 

With that phrace you will be saying the same as in line 61: the behaviour for `true` and there's not much said about the behaviour for `false`

Comment (ac-rappi) [reply README.md]: please re review

**Review by zeitos — APPROVED** [review]

**Pull request merged**